packages/@react-aria/overlays/src/ariaHideOutside.ts

```diff
@@ -73,6 +74,12 @@ export function ariaHideOutside(targets: Element[], root = document.body) {
     refCountMap.set(node, refCount + 1);
   };
 
+  // If there is already a MutationObserver listening from a previous call,
+  // disconnect it so the new on takes over.
+  if (observerStack.length) {
```

Comment:
This fixes nested dialog cases, where the MutationObservers would previously fight each other.